tests/readme.md

```diff
@@ -1,26 +1,41 @@
 
-Please use [pykale discussions on testing](https://github.com/pykale/pykale/discussions/categories/testing) to talk about tests and ask for help.
+## Quick start
 
-Refer to the [official pytest documentation](https://docs.pytest.org/en/stable/), or less formal [python testing software carpentry (alpha)](https://carpentries-incubator.github.io/python-testing/) and [tutorialspoint pytest tutorial](https://www.tutorialspoint.com/pytest/pytest_tutorial.pdf), if needed. This will help you to write tests and help with decisions on what aspects of your code need to be tested. There is some subjectivity involved in deciding how much of the potential behaviour of your code to check.
+- [Compact pytest tutorial](https://www.tutorialspoint.com/pytest/pytest_tutorial.pdf)
+- Example [unit tests for deep learning code of Variational Autoencoder (VAE) in PyTorch](https://github.com/tilman151/unittest_dl) and the associated post [How to Trust Your Deep Learning Code](https://krokotsch.eu/cleancode/2020/08/11/Unit-Tests-for-Deep-Learning.html). *Note: it uses `unittest` but we use `pytest`*.
+- Pytest+pytorch examples: [fastai1 tests](https://github.com/fastai/fastai1/tree/master/tests) and [Kornia tests](https://github.com/kornia/kornia/tree/master/test)
+- [Pytest fixtures](https://docs.pytest.org/en/stable/fixture.html), [Pytest exceptions](https://docs.pytest.org/en/stable/assert.html#assertions-about-expected-exceptions)
+- [Build-in torch testing modules](https://github.com/pytorch/pytorch/tree/master/torch/testing): `assert_tensors_equal` and `assert_tensors_allclose`
+- [fastai testing](https://fastai1.fast.ai/dev/test.html) is a good high-level reference. Its recommendations on [writing tests](https://fastai1.fast.ai/dev/test.html#writing-tests) are good to follow:
+  - Think about how to create a test of the real functionality that runs quickly, e.g. based on our `examples`.
+  - Use module scope fixtures to run initial code that can be shared amongst tests. When using fixtures, make sure the test doesn’t modify the global object it received, otherwise other tests will be impacted. If a given test modifies the global fixture object, it should either clone it or not use the fixture and create a fresh object instead.
+  - Avoid pretrained models, since they have to be downloaded from the internet to run the test.
+  - Create some minimal data for your test, or use data already in repo’s data/ directory.
+- Consider to use [pytest in vscode](https://code.visualstudio.com/docs/python/testing) or [pytest in pycharm](https://www.jetbrains.com/help/pycharm/pytest.html#pytest-bdd)
```

Comment:
Maybe a better link for pycharm is https://www.jetbrains.com/help/pycharm/pytest.html - while it looks interesting, I don't think we want to use BDD right now.